stackrox/stackrox
# ROX-17320: [offline v1] AlertHandler is now connection aware

## Description

AC sends alert results to sensor every time it blocks something. These alerts will be lost if the connection between central and sensor is broken. To avoid this, sensor responds to AC with an error if central is not reachable. AC alerts won't be lost since AC has a buffer that stores any alert that was not ACK'ed by sensor.

Note: The unit tests added are only meant to cover the online/offline switch behavior.

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

* [x] Manual testing with an infra cluster:
  1. Deploy ACS in an infra cluster
  2. Create a system policy that can be enforced by AC and enable AC to enforce objects on create.
  3. Break the connection between central and sensor making sensor go in offline mode
  4. Apply an object that will be blocked by AC
  5. Observe no violations reach central
  6. Reestablish connection between central and sensor
  7. Observe the violations reach central
* [x] Unit tested all components
